### ISSUE TITLE
Psalm skips `@var` tag over function call for constructor argument

### ISSUE THREAD

**Issue by andrew-demb** [vimeo/psalm#6298]
https://psalm.dev/r/7677857f64

```php
class SomeEvent {
    public function __construct(public \DateTimeImmutable $time) {}
}

function generateEvent(): SomeEvent {
    return new SomeEvent(
        /** @var \DateTimeImmutable */
    	convertValue('1970-01-01', 'dt_immutable')
    );
}
```

Expected: no errors, used type from provided `@var` tag over expression
Actual: INFO: MixedArgument - 10:6 - Argument 1 of SomeEvent::__construct cannot be mixed, expecting DateTimeImmutable

Full output
```
INFO: MixedArgument - 10:6 - Argument 1 of SomeEvent::__construct cannot be mixed, expecting DateTimeImmutable

  The type of convertValue('1970-01-01', 'dt_immutable') is sourced from here - 14:53
    function convertValue(string $value, string $type): mixed
```

PS: `convertValue` function cannot be templated because in real-world (`doctrine/dbal` - `Connection::convertToPHPValue()`) it is can be extended from any place with custom converters

**Comment by weirdan** [comment]
More generally, Psalm ignores docblocks on expressions
Related: #5599

**Comment by andrew-demb** [comment]
Is it a limitation from `nikic/parser`?
Does the described issue exist in the `nikic/parser` to share link for tracking?
Should it be documented somewhere in Psalm?

**Comment by andrew-demb** [comment]
Thank you for your answer.

Is it a limitation from `nikic/parser`?
Does the described issue exist in the `nikic/parser` to share link for tracking?
Should it be documented somewhere in Psalm?

**Comment by weirdan** [comment]
AFAIR the parser reports them just fine, it's Psalm the ignores them.
